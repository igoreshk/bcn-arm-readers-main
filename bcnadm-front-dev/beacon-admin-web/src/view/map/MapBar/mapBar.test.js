import React from 'react';
import { shallow } from 'enzyme';
import { MapBar } from './index';

const props = {
  levelNumber: 1,
  translate: jest.fn(),
  handleRemoveMarkers: jest.fn(),
  changeEditMode: jest.fn(),
  selectedMapProviderName: 'Google'
};

const wrapperWithoutBuildingSizeLegend = shallow(<MapBar {...props} />);
const wrapperWithBuildingSizeLegend = shallow(
  <MapBar {...props} buildingSizeForScaleMode={{ height: 100, width: 200 }} />
);

describe('MapBar', () => {
  it('should match snapshot', () => expect(wrapperWithoutBuildingSizeLegend).toMatchSnapshot());

  it('with buildingSizeForScaleMode prop should match snapshot', () =>
    expect(wrapperWithBuildingSizeLegend.first().shallow()).toMatchSnapshot());
});
