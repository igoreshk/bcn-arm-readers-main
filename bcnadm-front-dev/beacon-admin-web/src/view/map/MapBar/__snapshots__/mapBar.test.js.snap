// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapBar should match snapshot 1`] = `
<div>
  <div
    className="actions"
  >
    <div
      className="firstGroup"
    >
      <span
        className="levelNumber"
      />
      <withRouter(Connect(LocalizedComponent)) />
    </div>
    <div
      className="secondGroup"
    >
      <LocalizedComponent
        levelNumber={1}
        selectedMapProviderName="Google"
        translate={
          [MockFunction] {
            "calls": Array [
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
      <withRouter(ActionButtonsGroup)
        clear={[MockFunction]}
        toEdit={[MockFunction]}
        translate={
          [MockFunction] {
            "calls": Array [
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`MapBar with buildingSizeForScaleMode prop should match snapshot 1`] = `
<div>
  <div
    className="actions"
  >
    <div
      className="firstGroup"
    >
      <span
        className="levelNumber"
      />
      <withRouter(Connect(LocalizedComponent)) />
    </div>
    <div
      className="secondGroup"
    >
      <LocalizedComponent
        levelNumber={1}
        selectedMapProviderName="Google"
        translate={
          [MockFunction] {
            "calls": Array [
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
      <withRouter(ActionButtonsGroup)
        clear={[MockFunction]}
        toEdit={[MockFunction]}
        translate={
          [MockFunction] {
            "calls": Array [
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
              Array [
                "building.level",
                Object {
                  "value": 1,
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      />
    </div>
  </div>
  <LocalizedComponent
    height={100}
    width={200}
  />
</div>
`;
